### PR TITLE
Added grep actor

### PR DESCRIPTION
**Pull request by yarda**
Signed-off-by: Jaroslav Škarvada <jskarvad@redhat.com>

### CODE REVIEW

File: repos/system_upgrade/el7toel8/actors/grep/actor.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+                                'In locales with multibyte character encodings other than UTF-8, grep -P now reports an error and exits '
+                                'instead of misbehaving.\n'
+                                'When searching binary data, grep now may treat non-text bytes as line terminators. This can boost performance '
+                                'significantly.\n'
+                                'The \'grep -z\' no longer automatically treats the byte \'\200\' as binary data.\n'
+                                'Context no longer excludes selected lines omitted because of -m. For example, \'grep "^" -m1 -A1\' now outputs '
+                                'the first two input lines, not just the first line.\n',
+                        remediation='Please update your scripts to be compatible with the changes.',
+                        severity='low')
+                    break
```
Comment (fernflower) [inline repos/system_upgrade/el7toel8/actors/grep/actor.py]: iiuc you can return straightaway

File: repos/system_upgrade/el7toel8/actors/grep/actor.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    consumes = (InstalledRedHatSignedRPM,)
+    produces = (Report,)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        for fact in self.consume(InstalledRedHatSignedRPM):
+            for rpm in fact.items:
+                if rpm.name == 'grep':
+                    report_with_remediation(
+                        title='Grep incompatible changes in the next major version',
```
Comment (fernflower) [inline repos/system_upgrade/el7toel8/actors/grep/actor.py]: Missed has? 'Grep has incompatible..'

**Comment by yarda** [comment]
Hopefully all fixed.

File: repos/system_upgrade/el7toel8/actors/grep/actor.py (hunk 1 above)
Comment (yarda) [reply repos/system_upgrade/el7toel8/actors/grep/actor.py]: Fixed.

File: repos/system_upgrade/el7toel8/actors/grep/actor.py (hunk 2 above)
Comment (yarda) [reply repos/system_upgrade/el7toel8/actors/grep/actor.py]: Fixed, IIRC I took it from some other actor from the repo - I thought it's the way how it's written there, so I wanted to stay consistent.

**Comment by yarda** [comment]
I renamed the actor to checkgrep and fixed encoding problem which broke the tests on python2.

File: repos/system_upgrade/el7toel8/actors/checkgrep/actor.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+                                'the data to be non-matching.\n'
+                                'In locales with multibyte character encodings other than UTF-8, grep -P now reports an error and exits '
+                                'instead of misbehaving.\n'
+                                'When searching binary data, grep now may treat non-text bytes as line terminators. This can boost performance '
+                                'significantly.\n'
+                                'The \'grep -z\' no longer automatically treats the byte \'\\200\' as binary data.\n'
+                                'Context no longer excludes selected lines omitted because of -m. For example, \'grep "^" -m1 -A1\' now outputs '
+                                'the first two input lines, not just the first line.\n',
+                        remediation='Please update your scripts to be compatible with the changes.',
+                        severity='low')
```
Comment (fernflower) [inline repos/system_upgrade/el7toel8/actors/checkgrep/actor.py]: nit: I meant s/break/return, not removing that line :)

Comment (yarda) [reply repos/system_upgrade/el7toel8/actors/checkgrep/actor.py]: Sorry, sure, it may conserver few cycles.

Comment (yarda) [reply repos/system_upgrade/el7toel8/actors/checkgrep/actor.py]: Fixed.

**Pull request merged**